### PR TITLE
Feature/filepath

### PR DESCRIPTION
**Pull request by scottnath**
* adds new functions needed for files in content
* removes old content code, which was replaced by content middleware
* adds fixtures for files
* upgrades tests

---
relates to #588 

`DCO 1.1 Signed-off-by: Scott Nath <github@scottnath.com>`

### CODE REVIEW

File: lib/content/utils.js [hunk 1]
```diff
@@ -1,74 +1,95 @@
...
+
+  if (Array.isArray(files) && files.length > 0) {
+    files.forEach(filer => {
+      // if repeatable
+      if (Array.isArray(content[filer.attr][filer.input])) {
+        content[filer.attr][filer.input] = content[filer.attr][filer.input].map(atr => {
+          const attr = atr;
+
+          if (attr.value.hasOwnProperty('original') && attr.value.original !== '') {
+            attr.value.absolute = path.join(config.storage.public, attr.value.relative);
```
Comment (Snugug) [inline lib/content/utils.js]: * Why isn't this using [`storage.get`](https://github.com/punchcard-cms/punchcard/blob/master/lib/storage/get.js)
* Did you test this with HTTP or FTP urls? Things that should look like `method://path`? I'm fairly sure `path.join` destroys the double slash, which is required. That's why `storage.put` and `storage.get` work to normalize that so they can just be concatenated together.

File: lib/content/utils.js [hunk 2]
```diff
@@ -1,74 +1,95 @@
...
+          }
+
+          return attr;
+        });
+      }
+      else {
+        // non-repeatables
+        if (content[filer.attr] && content[filer.attr][filer.input].value.hasOwnProperty('original') && content[filer.attr][filer.input].value.original !== '') {
+          content[filer.attr][filer.input].value.absolute = path.join(config.storage.public, content[filer.attr][filer.input].value.relative);
```
Comment (Snugug) [inline lib/content/utils.js]: Why isn't this using `storage.get`?

Comment (Snugug) [inline lib/content/utils.js]: I don't see any tests to make sure that the actual content being spit out of here is actually correct. No stubs? No fixtures? Need tests for that

File: tests/apis.js [hunk 3]
```diff
@@ -44,7 +44,7 @@ test.serial('Utils: attributes - empty query', t => {
 
   return attributes.then(result => {
     t.is(typeof result, 'object', 'Should contain result, an object.');
-    t.is(Object.keys(result).length, 6, 'Should contain six main objects.');
+    t.is(Object.keys(result).length, 9, 'Should contain six main objects.');
```
Comment (Snugug) [inline tests/apis.js]: These descriptions are… wrong. Change the description or fix the test please

File: tests/content.js [hunk 4]
```diff
@@ -1,110 +1,86 @@
...
+  const result = utils.fileinputs(type.attributes);
 
-test('Check content id-not just a number', t => {
-  const rq = (JSON.parse(JSON.stringify(req)));
-  rq.params.id = 123;
-  const result = utils.check.id(rq);
-  t.false(result, 'Not in uuid');
-});
+  t.true(Array.isArray(result), 'should return an array');
+  t.is(result.length, 4, 'should have two items in array');
```
Comment (Snugug) [inline tests/content.js]: Two and four are not the same number

File: tests/content.js [hunk 5]
```diff
@@ -1,110 +1,86 @@
...
+  const type = futils.type('fileinputs');
+  const fileinputs = utils.fileinputs(type.attributes);
+  const result = utils.filepaths(fileinputs, futils.values('fileinputs'));
+
+  t.is(typeof result['fileinputs-file-single'], 'object', 'file-single input should be an object');
+  t.is(typeof result['fileinputs-file-single'].filesingle, 'object', 'file-single input should contain an input object');
+  t.is(typeof result['fileinputs-file-single'].filesingle.value, 'object', 'file-single input should contain an input value object');
+  t.is(typeof result['fileinputs-file-single'].filesingle.value.path, 'string', 'file-single input should contain an absolute string');
+  t.true(_.includes(result['fileinputs-file-single'].filesingle.value.path, config.storage.public), 'file-single input should contain the storage prefix');
+
```
Comment (Snugug) [inline tests/content.js]: I'd add some tests here to ensure that type and relative are in here as well, so it's not _just_ the public URL (to ensure that what's getting passed upwards has those)

File: tests/fixtures/files/index.js [hunk 6]
```diff
@@ -14,42 +13,6 @@ const all = [
 ];
 
 /*
- * Raw File Fixture
- *
- * @returns {array} - An array file objects as one receives from a multi-upload form
- */
-const raw = () => {
```
Comment (Snugug) [inline tests/fixtures/files/index.js]: Where did these fixtures disappear to?

Comment (scottnath) [inline tests/fixtures/files/index.js]: These are now in *shared tests*. Added [in this pr](https://github.com/punchcard-cms/shared-tests/pull/21)

Comment (scottnath) [inline tests/fixtures/files/index.js]: fwiw, I think that `saved` should move there too.

**Pull request merged**